tensorrt_llm/_torch/compilation/patterns/ub_allreduce.py

```
@@ -0,0 +1,526 @@
+import torch
+from torch._inductor.pattern_matcher import (CallFunction, Ignored, KeywordArg,
+                                             Match, MultiOutputPattern,
+                                             PatternMatcherPass, fwd_only,
+                                             register_replacement)
+
+import tensorrt_llm
+
+from ...distributed import AllReduceFusionOp, AllReduceStrategy
+
+aten = torch.ops.aten
+from tensorrt_llm.mapping import Mapping
```

Ruff (0.12.2)

Module level import not at top of file

(E402)